docs/core-concepts/queries.md

```diff
@@ -0,0 +1,148 @@
+
+```ts
+const messagesQuery = await query(getMessages, () => Date.now(), { interval: 30_000 })
+// automatically calls `getMessages` when query is created, and then again every 30 seconds
+```
+
+### Retries
+
+Automatically re-try the function when an error occurs. By default the query will wait `500` milliseconds between retries.
+
+```ts
+const connectionQuery = query(makeConnection, [token], { retries: 3 })
+// will try to call `makeConnection` a total of 3 times, waiting 500 MS between each attempt
+```
+
+Alternatively you can pass in `RetryOptions`, which allows you to specify both a `count` and a `delay`.
+
+```ts
+const retries = { count: 1, delay: 1_000 }
+const connectionQuery = query(makeConnection, [token], { retries: 3 })
+// will retry after 1,000 MS once if initial call fails
+```
+
+### Tags
+
+Tags are used to organize queries and mutations that are related. Tags can be used to access cached values, update cache values, and trigger cache invalidation.
+
+## UseQuery method
+
+The `useQuery` function adds some super useful, vue specific functionality you'd expect from a [composition](https://vuejs.org/guide/extras/composition-api-faq.html).
+
+### Parameters argument is reactive
+
+When parameters for your query function change, the query is automatically updated. The query is suspended while parameters are `null`, this enables you to create your query even when waiting on required arguments.
+
+```ts
+const selectedUserId = ref<string | undefined>()
+
+const userQuery = useQuery(fetchUser, () => selectedUserId.value ? [selectedUserId.value] : null)
+```
+
+### Delayed execution
+
+Queries will always wait if the parameters aren't ready, but you can also setup a query to always delay execution of the function until you either call `execute()` on the query, or change the reactive parameters argument.
+
+### Automatic cleanup
+
+When your component is unmounted, any queries are automatically disposed.
+
+## DefineQuery method
+
+```ts
+// Regular query - configure per usage
+const userQuery = query('user', fetchUser)
+const user1 = useQuery(userQuery, { params: [1], staleTime: 60000 })
+const user2 = useQuery(userQuery, { params: [2], staleTime: 120000 })
+
+// defineQuery - consistent configuration  
+const { useQuery: useUserQuery } = defineQuery(fetchUser, { staleTime: 60000 })
+const user1 = useUserQuery([1]) // Uses predefined staleTime
+const user2 = useUserQuery([2]) // Uses predefined staleTime
+```
+
+**Perfect for:**
+
+- **API layers** - Export both raw functions and pre-configured queries
+- **Shared query configuration** - Common settings across related queries  
+- **Component libraries** - Provide ready-to-use composables
+- **Team consistency** - Enforce caching and retry policies
+
+**Use regular `query` when:**
+
+- You need different options per usage
+- Simple one-off queries
+- You're using the default client (which doesn't have `defineQuery`)
```

Comment:
The heading is confusing because `query` is the name of the `query` function. 

The last bullet is false, the client isn't relavent here.